code/src/java/pcgen/core/display/CharacterDisplay.java

```diff
@@ -1087,15 +1080,10 @@ public boolean isSecondaryWeapon(final Equipment eq)
 			return false;
 		}
 
-		for (Equipment eqI : secondaryWeaponFacet.getSet(id))
-		{
-			if (eqI.getName().equalsIgnoreCase(eq.getName()) && (eqI.getLocation() == eq.getLocation()))
-			{
-				return true;
-			}
-		}
-
-		return false;
+		return secondaryWeaponFacet.getSet(id)
+		                           .stream()
+		                           .anyMatch(eqI -> eqI.getName().equalsIgnoreCase(eq.getName())
```

Comment:
Since this and line 1131 below are identical, should we break the equals items out into a separate method?  It would also reduce text in the stream, making it clearer.